lib/mysqlDaoQueryHelper.js

```diff
@@ -225,27 +225,32 @@ class MysqlDaoQueryHelper {
 
         properties.forEach(property => {
             const uncleanValue = obj[property];
-            const snakeCaseProperty = _.snakeCase(property);
+        const snakeCaseProperty = _.snakeCase(property);
 
-            // Escaping primitive values
-            if (_.includes(['string', 'number', 'boolean', 'undefined'], typeof(uncleanValue)) || uncleanValue === null) {
-                const cleanValue = config.dontCleanMysqlFunctions ? this.cleanSpecial(uncleanValue) : this.clean(uncleanValue) ;
-                cleanValues[snakeCaseProperty] = cleanValue;
-            }
-            // Escaping array values
-            else if (Array.isArray(uncleanValue)) {
-                const cleanValueArray = [];
-                uncleanValue.forEach(uncleanVal => {
-                    if (_.includes(['string', 'number', 'boolean'], typeof(uncleanVal))) {
-                        const cleanVal = config.dontCleanMysqlFunctions ? this.cleanSpecial(uncleanVal) : this.clean(uncleanVal) ;
-                        cleanValueArray.push(cleanVal)
-                    }
-                });
-                if (cleanValueArray.length > 0) {
-                    cleanValues[snakeCaseProperty] = cleanValueArray;
-                }
+        // Escaping primitive values
+        if (_.includes(['string', 'number', 'boolean', 'undefined'], typeof(uncleanValue)) || uncleanValue === null) {
+            const cleanValue = config.dontCleanMysqlFunctions ? this.cleanSpecial(uncleanValue) : this.clean(uncleanValue) ;
+            cleanValues[snakeCaseProperty] = cleanValue;
+        }
+        else if (_.isDate(uncleanValue)) {
+            //TODO: handle timezones somehow...
```

Comment:
From what I recall, the "clean" function is assigned a timezone from the application (defaulting to 'local'). Let me look into a config option real quick.